apis/v1alpha2/validation/gateway.go

```diff
@@ -44,7 +58,25 @@ func validateGatewaySpec(spec *gatewayv1a2.GatewaySpec, path *field.Path) field.
 // to the Gateway API specification.
 func validateGatewayListeners(listeners []gatewayv1a2.Listener, path *field.Path) field.ErrorList {
 	// TODO [danehans]: Add additional validation of listener fields.
-	return validateListenerHostname(listeners, path)
+	var errs field.ErrorList
+	errs = append(errs, validateListenerTLSConfig(listeners, path)...)
+	errs = append(errs, validateListenerHostname(listeners, path)...)
+	return errs
+}
+
+func validateListenerTLSConfig(listeners []gatewayv1a2.Listener, path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	for i, l := range listeners {
+		if isProtocolInSubset(l.Protocol, nonTlsProtocols) && l.TLS != nil {
+			errs = append(errs, field.Forbidden(path.Index(i).Child("tls"), fmt.Sprintf("should be empty for protocol %v", l.Protocol)))
+		}
+	}
```

Comment:
Sorry, I just realized that the TLS field should be populated for TLS-layered protocols and not for non-TLS protocols, so maybe something like this:

```suggestion
        for i, l := range listeners {        
                switch {    
                case isProtocolInSubset(l.Protocol, nonTlsProtocols):    
                        if l.TLS != nil {                                
                                errs = append(errs, field.Forbidden(path.Index(i).Child("tls"),    
                                        fmt.Sprintf("should be empty for protocol %v", l.Protocol)))    
                        }                                                                                                                               
                case isProtocolInSubset(l.Protocol, tlsProtocols):    
                        if l.TLS == nil {    
                                errs = append(errs, field.Forbidden(path.Index(i).Child("tls"),    
                                        fmt.Sprintf("should not be empty for protocol %v", l.Protocol)))    
                        }    
    
                }      
        }  
```

Reply:
Yep, I think there are valid reasons for TLS config to be empty even when the protocol is HTTPS or TLS. Between certs being attached through different means and controllers that may attach cert refs later, I don't think we can/should tighten the validation to require TLS config in those cases.